luigi/scheduler.py

```diff
@@ -591,6 +612,7 @@ def get_work(self, worker, host=None, **kwargs):
                               for worker in self._state.get_active_workers())
 
         tasks = list(self._state.get_pending_tasks())
+        self._state.cascade_disabled_state(tasks)
```

Comment:
> Is there a config setting that controls the number of times this can happen, and the delay for making them pending again?

No, the intention is actually that if a job flaps between failed and pending too often that it'll get disabled and persist in the disabled state for much longer. This can prevent retries from getting too nasty overnight. See http://luigi.readthedocs.org/en/latest/configuration.html#scheduler for the related parameters.

> How can you schedule a task as disabled and later schedule it as pending?

This happens if you set `disabled = True`, schedule a task, then set `disabled = False` and schedule it again. This is a common pattern if something causes a job to break and I need to disable it until I'm able to fix the conditions that are causing it to break. Once I set `disabled = False` again, I want the job and all its dependencies to be able to run again. The problem with your patch is that temporarily disabling a job would disable all of its downstream dependents until they're rescheduled, whether the disable was triggered within the scheduler or by scheduling it as disabled.

> Do you have an alternative suggestion for how to implement this?

I think you shouldn't disable tasks to prevent workers from being kept alive. If we want to let workers die when all of their tasks are disabled or upstream disabled, just cache the upstream disabled state and use that rather than marking them as disabled.